### ISSUE TITLE
'Customized Time Zone is an illegal timezone' error in Calendar Skill

### ISSUE THREAD

**Issue by ryub3n** [microsoft/botframework-components#1318]
## Describe the bug

Calendar Skill not able to fetch events because it's unable to recognize the working hours. 

## Version

Composer version 2.1.2
SDK version 4.15.2

## To Reproduce

1. Create the Enterprise Assistant with Calendar and People Skills.
2. Enter 'get my events'
3. See error 'Customized Time Zone is illegal timezone'

## Expected behavior

Should fetch the event details

## Screenshots

Error is in the below Dialog:
<img width="481" alt="image" src="https://user-images.githubusercontent.com/24279903/160610429-1dedc76d-24a0-4371-94f8-7b2d36c27270.png">

Error Message:
<img width="661" alt="image" src="https://user-images.githubusercontent.com/24279903/160610210-472407e6-2846-41a6-8dba-b82107222d4d.png">

Error Trace:
<img width="636" alt="image" src="https://user-images.githubusercontent.com/24279903/160610315-f1d11fa5-13d1-41ba-9a67-80880ab36f25.png">

**Comment by ramfattah** [comment]
Hello @ryub3n, I'm investigating this.

**Comment by ryub3n** [comment]
@ramfattah, I had manually changed my timezone to India in myaccount.microsoft.com
<img width="413" alt="image" src="https://user-images.githubusercontent.com/24279903/162556469-798ed8dc-0a96-4227-80a9-a7f1dff2b17f.png">
Could this change be causing the issue?

Here's the error log:

Category: Microsoft.Bot.Builder.Dialogs.Adaptive.Runtime.CoreBotAdapter
EventId: 0
ConnectionId: xxxxx
RequestId: xxxxx:xxxxx
RequestPath: /api/messages
SpanId: xxxxx
TraceId: xxxxx
ParentId: xxxxx
ActionId: xxxxx
ActionName: Calendar.Controllers.BotController.PostAsync (Calendar)

[OnTurnError] unhandled error : Customized Time Zone is an illegal timezone

Exception: 
System.Exception: Customized Time Zone is an illegal timezone
   at Microsoft.Bot.Components.Graph.GraphUtils.ConvertTimeZoneFormat(String timezone)
   at Microsoft.Bot.Components.Graph.Actions.RecognizeDateTime.BeginDialogAsync(DialogContext dc, Object options, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.DialogContext.BeginDialogAsync(String dialogId, Object options, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.Adaptive.Actions.ActionScope.BeginActionAsync(DialogContext dc, Int32 offset, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.Adaptive.Actions.ActionScope.OnNextActionAsync(DialogContext dc, Object result, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.Adaptive.Actions.ActionScope.ResumeDialogAsync(DialogContext dc, DialogReason reason, Object result, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.DialogContext.EndDialogAsync(Object result, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.Adaptive.Actions.SetProperties.BeginDialogAsync(DialogContext dc, Object options, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.DialogContext.BeginDialogAsync(String dialogId, Object options, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.Adaptive.Actions.ActionScope.BeginActionAsync(DialogContext dc, Int32 offset, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.Adaptive.Actions.ActionScope.OnNextActionAsync(DialogContext dc, Object result, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.Adaptive.Actions.ActionScope.ResumeDialogAsync(DialogContext dc, DialogReason reason, Object result, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.DialogContext.EndDialogAsync(Object result, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.Adaptive.Actions.TraceActivity.BeginDialogAsync(DialogContext dc, Object options, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.DialogContext.BeginDialogAsync(String dialogId, Object options, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.Adaptive.Actions.ActionScope.BeginActionAsync(DialogContext dc, Int32 offset, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.Adaptive.Actions.ActionScope.BeginDialogAsync(DialogContext dc, Object options, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.DialogContext.BeginDialogAsync(String dialogId, Object options, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.Adaptive.AdaptiveDialog.ContinueActionsAsync(DialogContext dc, Object options, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.Adaptive.AdaptiveDialog.BeginDialogAsync(DialogContext dc, Object options, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.DialogContext.BeginDialogAsync(String dialogId, Object options, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.Adaptive.Actions.BeginDialog.BeginDialogAsync(DialogContext dc, Object options, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.DialogContext.BeginDialogAsync(String dialogId, Object options, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.Adaptive.Actions.ActionScope.BeginActionAsync(DialogContext dc, Int32 offset, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.Adaptive.Actions.ActionScope.OnNextActionAsync(DialogContext dc, Object result, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.Adaptive.Actions.ActionScope.ResumeDialogAsync(DialogContext dc, DialogReason reason, Object result, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.DialogContext.EndDialogAsync(Object result, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.Adaptive.Actions.BeginDialog.ResumeDialogAsync(DialogContext dc, DialogReason reason, Object result, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.DialogContext.EndDialogAsync(Object result, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.Adaptive.AdaptiveDialog.OnEndOfActionsAsync(ActionContext actionContext, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.Adaptive.AdaptiveDialog.ContinueActionsAsync(DialogContext dc, Object options, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.Adaptive.AdaptiveDialog.ContinueDialogAsync(DialogContext dc, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.DialogContext.ContinueDialogAsync(CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.Adaptive.AdaptiveDialog.ContinueActionsAsync(DialogContext dc, Object options, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.Adaptive.AdaptiveDialog.ContinueDialogAsync(DialogContext dc, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.DialogContext.ContinueDialogAsync(CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.DialogExtensions.InnerRunAsync(ITurnContext turnContext, String dialogId, DialogContext dialogContext, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.DialogExtensions.InternalRunAsync(ITurnContext turnContext, String dialogId, DialogContext dialogContext, DialogStateManagerConfiguration stateConfiguration, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.DialogExtensions.InternalRunAsync(ITurnContext turnContext, String dialogId, DialogContext dialogContext, DialogStateManagerConfiguration stateConfiguration, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.DialogExtensions.RunAsync(Dialog dialog, ITurnContext turnContext, IStatePropertyAccessor`1 accessor, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Dialogs.Adaptive.AdaptiveDialogBot.OnTurnAsync(ITurnContext turnContext, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.SetSpeakMiddleware.OnTurnAsync(ITurnContext turnContext, NextDelegate next, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.TranscriptLoggerMiddleware.OnTurnAsync(ITurnContext turnContext, NextDelegate nextTurn, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.TelemetryLoggerMiddleware.OnTurnAsync(ITurnContext context, NextDelegate nextTurn, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Integration.ApplicationInsights.Core.TelemetryInitializerMiddleware.OnTurnAsync(ITurnContext context, NextDelegate nextTurn, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.MiddlewareSet.ReceiveActivityWithStatusAsync(ITurnContext turnContext, BotCallbackHandler callback, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.BotAdapter.RunPipelineAsync(ITurnContext turnContext, BotCallbackHandler callback, CancellationToken cancellationToken)

**Comment by ramfattah** [comment]
@ryub3n 

Are you getting this error `Customized Time Zone is illegal timezone` in emulator only? 

Have you had a chance to try with other channels to see if you are reproducing the same issue?

**Comment by ryub3n** [comment]
@ramfattah , I've tried in other channels as well, the same issue.

**Comment by ramfattah** [comment]
@ryub3n 

Thank you for your patience.
#

> I had manually changed my timezone to India in myaccount.microsoft.com
> Could this change be causing the issue?

I'm not totally sure, have you tried to reset your timezone back to the previous timezone?

**Comment by ramfattah** [comment]
update: 

Looks like this is the line code where it is throwing this error:

https://github.com/microsoft/botframework-components/blob/da8fb3587ab99dbff39b3475049d25b91203ce08/packages/Graph/GraphUtils.cs#L32

**Comment by ryub3n** [comment]
@ramfattah, I managed to resolve this issue but I think it's an outlook bug.
I checked the bot and saw that the Graph package uses and converts **user.workinghours.timezone**
<img width="334" alt="image" src="https://user-images.githubusercontent.com/24279903/163467463-9520e62b-98d0-4bce-9c94-c4019f087b8b.png">

I searched and found out that workingHours is set in mailbox which led me to Outlook web.
Outlook web was showing this and I don't know why:
![image](https://user-images.githubusercontent.com/24279903/163467298-fb6f4581-d3cc-431c-a5ac-46b189aec814.png)

So I clicked on the checkbox below that said 'I'm going to remain...' and tested the bot and it worked.
<img width="672" alt="image" src="https://user-images.githubusercontent.com/24279903/163467114-42bfb9fd-2a79-446c-820f-b5d9f98b04c4.png">
